examples/wordCount/lines2words/lines2words.cpp

```diff
@@ -27,69 +27,29 @@ void signalCallbackHandler(int signum) {
-      auto line = receive();
-      measurementHandler->setField(std::to_string(computeCounter), "CMP_START",
-                                   0);
-      std::stringstream streamedLines(line);
-      std::string word;
-      while (streamedLines >> word) {
-        std::cout << word << std::endl;
-        push(word);
-        measurementHandler->setField(std::to_string(computeCounter),
-                                     "lines1->words", 0);
-        measurementHandler->setField(std::to_string(computeCounter),
-                                     "CMP_FINISH", 0);
-      }
+  void lines2words(const std::string &line,
+                   std::function<void(std::vector<uint8_t>)> push) {
+    measurementHandler->setField(std::to_string(m_computeCounter), "CMP_START",
+                                 0);
+    std::stringstream streamedLines(line);
+    std::string word;
+    while (streamedLines >> word) {
+      std::vector<uint8_t> data(word.begin(), word.end());
+      push(data);
+      measurementHandler->setField(std::to_string(m_computeCounter),
+                                   "lines1->words", 0);
+      measurementHandler->setField(std::to_string(m_computeCounter),
+                                   "CMP_FINISH", 0);
+      m_computeCounter++;
     }
   }
-};
-
-void run(const std::string &syncPrefix, const std::string &nodePrefix,
-         const std::string &subTopic, const std::string &pubTopic,
-         uint32_t numberOfProducerPartitions,
-         std::vector<uint32_t> consumerPartitions,
-         std::chrono::milliseconds publishInterval) {
-  WordSplitter wordSplitter;
-  ndn::Face face;
-  auto iceflow =
-      std::make_shared<iceflow::IceFlow>(syncPrefix, nodePrefix, face);
-  auto producer = iceflow::IceflowProducer(
-      iceflow, pubTopic, numberOfProducerPartitions, publishInterval);
-  auto consumer =
-      iceflow::IceflowConsumer(iceflow, subTopic, consumerPartitions);
-
-  std::vector<std::thread> threads;
-  threads.emplace_back(&iceflow::IceFlow::run, iceflow);
-  threads.emplace_back([&wordSplitter, &consumer, &producer]() {
-    wordSplitter.lines2words(
-        [&consumer]() -> std::string {
-          auto data = consumer.receiveData();
-          return std::string(data.begin(), data.end());
-        },
-        [&producer](const std::string &data) {
-          std::vector<uint8_t> encodedString(data.begin(), data.end());
-          producer.pushData(encodedString);
-        });
-  });
-
-  for (auto &thread : threads) {
-    thread.join();
-  }
-}
 
-std::string generateNodePrefix() {
-  boost::uuids::basic_random_generator<boost::mt19937> gen;
-  boost::uuids::uuid uuid = gen();
-  return to_string(uuid);
-}
+private:
+  int m_computeCounter = 0;
+};
 
-std::vector<uint32_t> createConsumerPartitions(iceflow::Edge upstreamEdge,
+std::vector<uint32_t> createConsumerPartitions(uint32_t maxConsumerPartitions,
                                                uint32_t consumerIndex) {
```

Comment:
The naming of these variables is a bit confusing, as it isn't clear that `maxConsumerPartitions` denotes the highest partition number, not the number of partitions relative to `consumerIndex`.

Maybe calling these `partitionStart` and `partitionEnd` or something along those lines would be better.

Reply:
Thanks for noticing that, the implementation of that function was actually kind of broken 😅 I now tried to improve it a bit via 58737c79850e551de0db79efb25d30cab599a9b7